src/v/cluster/security_manager.cc

```diff
@@ -153,8 +153,13 @@ security_manager::fill_snapshot(controller_snapshot& controller_snap) const {
 
     for (const auto& cred : _credentials.local()) {
         ss::visit(cred.second, [&](security::scram_credential scram) {
-            snapshot.user_credentials.push_back(user_and_credential{
-              security::credential_user{cred.first}, std::move(scram)});
+            if (
+              scram.principal().has_value()
+              && scram.principal()->type()
+                   != security::principal_type::ephemeral_user) {
+                snapshot.user_credentials.push_back(user_and_credential{
+                  security::credential_user{cred.first}, std::move(scram)});
+            }
```

Comment:
@bharathv or anyone from the cluster team, could you review this commit to make sure it is OK since this file is in the cluster sub-system?

Reply:
https://redpandadata.slack.com/archives/C04B5T58QQ1/p1688129975646029?thread_ts=1688128227.519959&cid=C04B5T58QQ1